### ISSUE TITLE
Group Policy is blocking SAGA-functions

### ISSUE THREAD

**Issue by c-m-mueller** [qgis/QGIS#30250]
**The bug**

In our company, we uses the group policy objects to restrict the access to different folders in our network. Unfortunately, this also applies to the temporary folders "temp" and "tmp". When using different SAGA-functions (yet, tested with "inverse distance weighted interpolation" and "ordinary kriging") I get the following error-message:

![Unbenannt](https://user-images.githubusercontent.com/51915837/59607874-716f3e00-9114-11e9-8add-6ce9e9df5321.png)

The marked message means, that the programm "statistics_kriging "Ordinary Kriging"" has been blocked due our group policy. According to our administrator this problem is caused, when QGIS is coping the statistics_kriging.dll to the temporary folder to run it there. 

We have no option to change the group policy. 

My first idea was to change the temporary folder (for QGIS, not for the whole system!). Is there any option to change the temporary folders for QGIS an the SAGA, GRASS and GDAL functions?

**How to Reproduce**

We have reproduced this problem on different computers in the company with the same problem on every computer. 

**QGIS and OS versions**

QGIS version: 3.4.7 - Madeira (LTR)
OS version: WIN 7 Professional - 6.1.7601 Service Pack 1 (x64)

**Comment by gioman** [comment]
> My first idea was to change the temporary folder (for QGIS, not for the whole system!). Is there any option to change the temporary folders for QGIS an the SAGA, GRASS and GDAL functions?

I guess this would e a feature request.

**Comment by nyalldawson** [comment]
There's no such option, and honestly, it's unlikely that one will be added for this specific use case. It seems to me that this policy is always going to cause issues - even if we allow the processing temporary directory to be user settable, it's highly likely you'll hit the same issue in other parts of qgis, saga and grass.

I'd suggest revisiting this discussion with your IT provider.

**Comment by nyalldawson** [comment]
I also should point out that qgis isn't copying any DLL to the temp folder - if that's happening, it's happening from within Saga.

**Comment by alexbruy** [comment]
QGIS does not copy any DLL's to the temp folder, as well as SAGA. I'm with Nyall here, this is a policy issue, not QGIS. BTW, temporary directory used by QGIS can be easily changed by setting `TMP` and/or `TEMP` environment values.

Closing as wontfix.

**Comment by stragu** [comment]
Following up on this, I've got a quick question:

I am using QGIS 3.16.4 on Windows 10, at a University, and I noticed that SAGA and GRASS tools are blocked by group policy when used from QGIS, but the exact same analysis run directly form SAGA or GRASS is not blocked.

How is that possible, if they are using the exact same executables?

**Comment by tilmanb** [comment]
> Following up on this, I've got a quick question:
> 
> I am using QGIS 3.16.4 on Windows 10, at a University, and I noticed that SAGA and GRASS tools are blocked by group policy when used from QGIS, but the exact same analysis run directly form SAGA or GRASS is not blocked.
> 
> How is that possible, if they are using the exact same executables?

Often batch files are blocked by group policy. That means the SAGA executables are fine and allowed, but the batch file the QGIS plugin writes to call the SAGA commands is blocked.

**Comment by tilmanb** [comment]
SAGA-NG 1.1.0 ( https://plugins.qgis.org/plugins/processing_saga_nextgen/version/1.1.0/ ) has the possibility to write the scripts to a user-specified location, which can be whitelisted for applocker.

**Comment by stragu** [comment]
Thanks for the heads up and your contribution! Linking to the relevant commit in case people are interested: https://github.com/north-road/qgis-processing-saga-nextgen/commit/e69ec53426091e9e790cd94bacfe6e9fec847cc5
